src/9FC00B40.c

```diff
@@ -39,27 +117,387 @@ s32 skGetId(BbId* id) {
     return 0;
 }
 
-INCLUDE_ASM("asm/non_matchings/9FC00B40", func_9FC00DCC);
+s32 func_9FC00DCC(BbTicketBundle* bundle, BbAppLaunchCrls* crls, RecryptList* recryptList, s32 a3) {
+    s32 ret;
+
+    if (!check_ticket_bundle_ranges(bundle)) {
+        return -1;
+    }
+
+    if (a3 && (bundle->ticket->cmd.head.execFlags & 2) == 0) {
+        return 1;
+    }
+
+    if (!CHECK_UNTRUSTED(crls) || !CHECK_UNTRUSTED_RECRYPT_LIST(recryptList)) {
+        return -1;
+    }
+
+    ret = check_ticket_bundle_revocations(bundle, crls);
+
+    if (ret == 0) {
+        ret = func_9FC00BAC(bundle);
+
+        if (ret == 0) {
+            return 0;
+        }
+    }
+
+    return ret;
+}
+
+s32 skLaunchSetup(BbTicketBundle* bundle, BbAppLaunchCrls* crls, RecryptList* recryptList) {
+    BbAesKey sp10;
+    BbAesKey* key;
+    BbTicketHead* head;
+    s32 ret;
+
+    ret = func_9FC00DCC(bundle, crls, recryptList, FALSE);
+
+    if (ret) {
```

Comment:
```suggestion
    ret = func_9FC00DCC(bundle, crls, recryptList, FALSE);
    if (ret != 0) {
```
Since `func_9FC00DCC` does not return boolean values I think it's slightly clearer in intention to write the `!= 0` explicitly